### PR TITLE
Update Jenkinsfile to implement slack notifications

### PR DESCRIPTION
**Pull request by Nazgolze**
### What was the problem?
Jenkinsfile lacked slack integration
### How did I fix it?
I added it
### How to test it?
Conveniently by pushing this branch to github (and doing a few replay builds to force the error condition)
### Review checklist
- The PR solves #583 
- All new code is covered with unit tests
- All new features are covered with e2e tests
- All new code follows best practices

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -79,6 +119,12 @@ pipeline {
 		}
 	}
 	post {
+		success {
+			recovery()
```
Comment (fchavant) [inline Jenkinsfile]: Move the content of the `recovery` function here for clarity (it's only called here)

File: Jenkinsfile [hunk 2]
```diff
@@ -79,6 +119,12 @@ pipeline {
...
+		}
+		failure {
+			fail()
```
Comment (fchavant) [inline Jenkinsfile]: Move the content of the `fail` function here for clarity (it's only called here)

File: Jenkinsfile [hunk 3]
```diff
@@ -1,3 +1,43 @@
...
+  if ( channel.size() > CHANNEL_MAX_LEN ) {
+     channel = channel.substring(0, CHANNEL_MAX_LEN)
+  }
+
+  echo "[slack_send] channel: ${channel} "
+  slackSend color: "${color}", message: "${message}", channel: "${channel}"
+}
+
+def recovery() {
+  if (currentBuild.result == null || currentBuild.result == 'SUCCESS') {
```
Comment (fchavant) [inline Jenkinsfile]: Do we want notifications only on recovery? This was a requirements for some other project, not sure about lisk-explorer though.

File: Jenkinsfile [hunk 4]
```diff
@@ -1,3 +1,43 @@
...
+      build_info = get_build_info()
+      slack_send('good', "Recovery: build ${build_info} was successful.")
+    }
+  }
+}
+
+def fail() {
+  build_info = get_build_info()
+  slack_send('danger', "Build ${build_info} failed (<${env.BUILD_URL}/console|console>, <${env.BUILD_URL}/changes|changes>)\n")
+  currentBuild.result = 'FAILURE'
```
Comment (fchavant) [inline Jenkinsfile]: Superfluous: if we reach this line then the build has already failed.

File: Jenkinsfile (hunk 3 above)
Comment (Nazgolze) [reply Jenkinsfile]: This is a good question

Comment (Nazgolze) [reply Jenkinsfile]: @MichalTuleja How do we want this to behave?

File: Jenkinsfile [hunk 5]
```diff
@@ -79,6 +102,23 @@ pipeline {
 		}
 	}
 	post {
+		success {
+			script {
+				if (currentBuild.result == null || currentBuild.result == 'SUCCESS') {
```
Comment (fchavant) [inline Jenkinsfile]: Shouldn't this read `currentBuild.result != null`?

**Comment by MichalTuleja** [comment]
@fchavant  @Nazgolze These changes will be also applied to the `1.6.1` branch. Is there anything that should be changed after the cherry-picking?

**Comment by Nazgolze** [comment]
@MichalTuleja I don't think so. There's nothing in this change that's specific to a particular version of lisk-explorer. I guess the test then is to see if you get a slack notification after cherry-picking and which one it is ;-)

**Pull request merged**